ark/phenotyping/preprocess.py

```diff
@@ -10,6 +11,60 @@
 from ark.utils import load_utils
 from ark.utils import misc_utils
 
+from timeit import default_timer
+
+
+def _unpacking_apply_along_axis(func_args):
+    """Maps apply_along_axis to each array subset
+
+    Args:
+        func_args (tuple):
+            The arguments to pass into apply_along_axis
+
+    Returns:
+        numpy.ndarray:
+            The result for apply_along_axis for a specific array subset
+    """
+
+    (func1d, axis, arr, args, kwargs) = func_args
+    return np.apply_along_axis(func1d, axis, arr, *args, **kwargs)
+
+
+def _parallel_apply_along_axis(func1d, axis, arr, *args, **kwargs):
+    """Runs apply_along_axis in parallel for a massive speedup (but at a massive memory cost too)
+
+    Args:
+        func1d (Callable):
+            The function to apply along the given axis
+        axis (int):
+            The axis to run
+
+    Returns:
+        numpy.ndarray:
+            The processed array, equivalent to what np.apply_along_axis returns
+    """
+
+    # If axis is 0 we'll effectively be working with the 1st axis
+    effective_axis = 1 if axis == 0 else axis
+
+    # Swap the 0th and 1st axis if axis specified is 0
+    if effective_axis != axis:
+        arr = arr.swapaxes(axis, effective_axis)
+
```

Comment:
idk why we're doing this, but if we need this then we can just do:
```
if axis == 0:
    arr = arr.swapaxes(0, 1)
    axis = 1
```

Reply:
I'll probably just end up removing this. I think this was something specific the user needed but we don't.